controllers/datamovement_controller.go

```diff
@@ -214,40 +254,154 @@ func (r *DataMovementReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	go func() {
 		var cmd *exec.Cmd
 
-		// TODO: UserId/GroupId
-
-		if isTestEnv() {
-			cmd = exec.CommandContext(ctxCancel, "sleep", "1")
+		// Allow the data movement command to be overridden via the configmapa
+		// for testing purposes
+		if dmCmd != "" {
```

Comment:
I wonder if we hide this complexity introduced via the config and instead use functions
```go
func getCommand(dm *nnfv1alpha1.DataMovement, config configMap) string {
  if len(config[commandOverride]) == 0 {
    return strings.Split(config[commandOverride], " ")[0]
  }
  
  return "mpirun"
}

func getArguments(dm *nnfv1alpha1.DataMovement, config configMap) []string {
  if len(config[commandOverride]) == 0 {
    return strings.Split(config[commandOverride], " ")[1:]
  }
  // ...
}
```

Reply:
👍